x/incentives/keeper/distribute.go

```diff
@@ -713,6 +698,28 @@ func (k Keeper) distributeInternal(
 	return totalDistrCoins, err
 }
 
+func (k Keeper) skipSpamGaugeDistribute(ctx sdk.Context, locks []*lockuptypes.PeriodLock, gauge types.Gauge, totalDistrCoins sdk.Coins, remainCoins sdk.Coins) (bool, sdk.Coins, error) {
```

Comment:
I think it'd be worth creating unit test for this as well

Reply:
Added https://github.com/osmosis-labs/osmosis/pull/7250/commits/5cf383a0df80454fad9f4c202f02dc903c9fef3b